roles/network-multus/tasks/provision.yml

```diff
@@ -46,24 +46,83 @@
 
   - name: Create multus Resources
     shell: "{{ cluster_command }} apply -f /tmp/multus.yml"
+  when: crd.stdout=="0"
 
+- name: Deploy CNI plugins
+  block:
   - name: Render cni plugins deployment yaml
     template:
       src: cni-plugins.yml
       dest: /tmp/cni-plugins.yml
 
   - name: Create cni plugins Resources
     shell: "{{ cluster_command }} apply -f /tmp/cni-plugins.yml"
+  when: deploy_cni_plugins
+
+- name: Deploy OVS Plugin
+  block:
+  - name: Render OVS deployment yaml
+    template:
+      src: ovs-cni.yml
+      dest: /tmp/ovs-cni.yml
 
+  - name: Create ovs Resources
+    shell: "{{ cluster_command }} apply -f /tmp/ovs-cni.yml"
 
-  - name: Wait until multus is running
-    shell: "{{ cluster_command }} -n {{ namespace }} get daemonset | grep {{ item }} | awk \'{ if ($3 == $4) print \"0\"; else print \"1\"}\'"
-    loop:
-    - kube-multus-ds-amd64
-    - kube-cni-plugins-ds-amd64
-    register: result
-    until: result.stdout == "0"
-    retries: 20
-    delay: 5
+  - name: Deploy ovs-vsctl template
+    block:
+    - name: check if ovs is installed
+      yum:
+        list="openvswitch"
+      register: ovs_pkgs
 
-  when: crd.stdout=="0"
+    - name: fail if ovs is installed
```

Comment:
Will this block prevent this playbook from being idempotent?  Why should the playbook fail if it's installed?

Reply:
@phoracek could you please elaborate on why we need to fail if ovs is already installed on openshift env?